src/lib/angles.js

```diff
@@ -10,25 +10,241 @@
 
 var PI = Math.PI;
 
-exports.deg2rad = function(deg) {
+function deg2rad(deg) {
     return deg / 180 * PI;
-};
+}
 
-exports.rad2deg = function(rad) {
+function rad2deg(rad) {
     return rad / PI * 180;
-};
+}
 
-exports.wrap360 = function(deg) {
+function wrap360(deg) {
     var out = deg % 360;
     return out < 0 ? out + 360 : out;
-};
+}
 
-exports.wrap180 = function(deg) {
+function wrap180(deg) {
     if(Math.abs(deg) > 180) deg -= Math.round(deg / 360) * 360;
     return deg;
-};
+}
 
-exports.isFullCircle = function(sector) {
+/**
+ * is sector a full circle?
+ * ... this comes up a lot in SVG path-drawing routines
+ *
+ * @param {2-item array} sector sector angles in *degrees*
+ * @return {boolean}
+ */
+function isFullCircle(sector) {
     var arc = Math.abs(sector[1] - sector[0]);
     return arc === 360;
+}
+
+/**
+ * angular delta between angle 'a' and 'b'
+ * solution taken from: https://stackoverflow.com/a/2007279
+ *
+ * @param {number} a : first angle in *radians*
+ * @param {number} b : second angle in *radians*
+ * @return {number} angular delta in *radians*
+ */
+function angleDelta(a, b) {
+    var d = b - a;
+    return Math.atan2(Math.sin(d), Math.cos(d));
+}
+
+/**
+ * angular distance between angle 'a' and 'b'
+ *
+ * @param {number} a : first angle in *radians*
+ * @param {number} b : second angle in *radians*
+ * @return {number} angular distance in *radians*
+ */
+function angleDist(a, b) {
+    return Math.abs(angleDelta(a, b));
+}
+
+/**
+ * is angle inside sector?
+ *
+ * @param {number} a : angle to test in *radians*
+ * @param {2-item array} sector : sector angles in *degrees*
```

Comment:
In degrees here as before this PR this routine was only used with `layout.polar.sector` as argument which is always in degrees.

But I'm thinking of standardizing all angles in this file to be in radians.

Reply:
> But I'm thinking of standardizing all angles in this file to be in radians.

That sounds like a good idea, would probably simplify quite a few things. BTW `wrap360` is really just `Lib.mod`, right? And I'm thinking perhaps that we want a version of `Lib.mod` that restricts to +/- 1/2 the value, rather than 0-1x as `Lib.mod` does now - which would then cover both `angleDelta` and `wrap180`.

Reply:
Standardized to radians in https://github.com/plotly/plotly.js/pull/2954/commits/e93897473bec9cf27f7e4a63cf85ce55ac4de1b6



> BTW wrap360 is really just Lib.mod, right? And I'm thinking perhaps that we want a version of Lib.mod that restricts to +/- 1/2 the value, rather than 0-1x as Lib.mod does now - which would then cover both angleDelta and wrap180.

I'll have to look at this in more detail :thinking:

Reply:
In https://github.com/plotly/plotly.js/pull/2954/commits/8575b27ad369921ab7e842493fb68d6fb566efea, I replaced `wrap360` with its `mod` equivalent - which turns out well as we no longer need to convert thing to degrees `mod(v, 2 * Math.PI)` works just fine.

In the same commit, I added a "restricts to +/- 1/2 the value" version of `mod` that I named `Lib.modHalf` (please let me know if you can think of a better name for this thing).